### ISSUE TITLE
Fail to process batch data using EMD loss

### ISSUE THREAD

**Issue by zhangmozhe** [justanhduc/neuralnet-pytorch#4]
Hi, I met error "The size of tensor a (2048) must match the size of tensor b (4096) at non-singleton dimension 1" when I run the following code:

points1 = torch.rand(2, 2048, 3, requires_grad=True).cuda()
points2 = torch.rand(2, 2048, 3).cuda()
geomloss.SamplesLoss(loss='sinkhorn', p=2, blur=0.05)(points1 , points2)

It seems that the code does not process one batch input well. I wonder how to solve it. Thanks!

**Comment by justanhduc** [comment]
Hello @zhangmozhe. First of all, I am not sure if you should ask me about ``geomloss`` here, as I do not maintain it.
But anw, to answer your question, could you please try setting the [`backend`](https://github.com/jeanfeydy/geomloss/blob/4e09e3bfd376d92f2bb7efdf0854a5b7c756eb0d/geomloss/samples_loss.py#L157) argument to `tensorized` explicitly?
Also, just fyi in case you miss it, EMD and Sinkhorn have slightly different formulation. Please beware of it so you don't mistakenly use this function.

**Comment by zhangmozhe** [comment]
Hi @justanhduc, thanks for your reply. I wonder what's the problem when using Sinkhorn? I had thought Sinkhorn is a fast approximation of EMD. Thanks for your help very much!

**Comment by justanhduc** [comment]
Hello. Yes and no. Yes because it is an iterated method, but no because Sinkhorn has an entropy regularization term, and thanks to the regularization, it has a nice solution like that. Just keep that in mind and consider whether this regularization is significant to your problem or not. Probably I was quite irresponsible when including them under one roof like that.
